packages/eas-cli/src/commands/secrets/create.ts

```diff
@@ -0,0 +1,153 @@
+      options: ['account', 'project'],
+    },
+    {
+      name: 'name',
+      required: false,
+      description: 'Name of the secret',
+    },
+    {
+      name: 'value',
+      required: false,
+      description: 'Value of the secret',
+    },
+  ];
+
+  async run() {
+    const user = await ensureLoggedInAsync();
+    let {
+      args: { name, value: secretValue, target },
+    } = this.parse(EnvironmentSecretCreate);
+
+    const projectDir = (await findProjectRootAsync()) ?? process.cwd();
+    const accountName = await getProjectAccountNameAsync(projectDir);
+
+    const {
+      exp: { slug },
+    } = getConfig(projectDir, { skipSDKVersionRequirement: true });
+
+    const projectId = await ensureProjectExistsAsync({
+      accountName,
+      projectName: slug,
+    });
+
+    if (!(await isEasEnabledForProjectAsync(projectId))) {
+      warnEasUnavailable();
+      process.exitCode = 1;
+      return;
+    }
+
+    if (!target) {
+      const validationMessage = 'Secret target may not be empty.';
+
+      ({ target } = await promptAsync({
+        type: 'select',
+        name: 'target',
+        message: 'Where should this secret be used:',
+        choices: [
+          { title: 'Account-wide', value: 'account' },
+          { title: 'Project-specific', value: 'project' },
+        ],
+        validate: value => (value ? true : validationMessage),
+      }));
+    }
+
+    if (!name) {
+      ({ name } = await promptAsync({
+        type: 'text',
+        name: 'name',
+        message: `Secret name:`,
+        validate: value => {
+          if (!value) {
+            return 'Secret name may not be empty.';
+          }
+
+          if (value.match(/^\w+$/)) {
+            return 'Names may contain only letters, numbers, and underscores.';
+          }
+
+          return true;
+        },
+      }));
+    }
+
+    if (!secretValue) {
+      const validationMessage = 'Secret value may not be empty.';
+
+      ({ secretValue } = await promptAsync({
+        type: 'text',
+        name: 'secretValue',
+        message: 'Secret value:',
+        validate: value => (value ? true : validationMessage),
+      }));
+    }
+
+    let secret;
+
+    if (target === 'project') {
+      if (!projectDir) {
+        throw new Error('Please run this command inside a project directory.');
+      }
+
+      secret = await EnvironmentSecretMutation.createForApp(
+        { name, value: secretValue },
+        projectId
+      );
+      if (!secret) {
+        throw new Error(
+          `Could not create secret with name ${name} on project with id ${projectId}`
+        );
+      }
+
+      Log.withTick(
+        `️Created a new secret ${chalk.bold(name)} on project ${chalk.bold(
+          `@${accountName}/${slug}`
+        )}.`
+      );
+    }
+
+    if (target === 'account') {
+      const resolver = new AccountResolver(projectDir, user);
+      const ownerAccount = await resolver.resolveAccountAsync();
```

Comment:
you already know `accountName` then you can just find is with findAccountByName from https://github.com/expo/eas-cli/blob/main/packages/eas-cli/src/user/Account.ts